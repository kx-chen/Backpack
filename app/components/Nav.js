import React from 'react';
import SubredditIcon from './SubredditIcon';

function Nav({ onMenuChange, open, onSubredditSelect, selected_subreddit, subreddits, downloadSubreddit }) {
  let width = '78px';
  if (open) {
    width = '200px';
  }

  let left = '85px';
  if (open) {
    left = '220px';
  }

  const arrowStyle = {
    left: left,
    position: 'absolute',
    transition: '0.3s'
  };

  const sidebarStyle = {
    width: width
  };

  const subredditsList = subreddits.map(subreddit => (
    <SubredditIcon name={subreddit.name}
                   open={open}
                   onSelected={ () => onSubredditSelect(subreddit)}
                   downloadSubreddit={(name) => downloadSubreddit(name)}
                   selected={subreddit.name === selected_subreddit}/>
  ));

  return (
    <nav className="sidebar" style={sidebarStyle}>
      <div className="nav-item search">
        <img
          alt="search"
          src="../resources/search.svg"
          height="27"
          className="nav-clickable"
        />
      </div>
      {subredditsList}
      <div className="nav-item--bottom">
        <img
          alt="settings"
          src="../resources/cog.svg"
          height="30"
          className="nav-clickable"
        />
        <img
          alt="open and close menu"
          src="../resources/arrow-circle-left.svg"
          height="30"
          className="nav-clickable"
          style={arrowStyle}
          onClick={() => onMenuChange(open)}
        />
      </div>
    </nav>
  );
}

export default Nav;
