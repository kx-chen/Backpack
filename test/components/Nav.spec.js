import { spy } from 'sinon';
import React from 'react';
import Enzyme, { shallow } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';

import Nav from '../../app/components/Nav';

Enzyme.configure({ adapter: new Adapter() });

function setup() {
  const actions = {
    onMenuChange: spy(),
    onSubredditSelect: spy(),
    onDownloadSubreddit: spy(),
    onSaveNewSubreddit: spy(),
  };
  const component = shallow(<Nav open={false} subreddits={[]} selectedSubreddit={''} {...actions} />);
  return {
    component,
    actions,
    menuToggle: component.find('#open-close-menu'),
  };
}

describe('Nav component', () => {
  it('should call onMenuChange on click', () => {
    const { menuToggle, actions } = setup();
    menuToggle.simulate('click');
    expect(actions.onMenuChange.called).toBe(true);
  });

  it('should match exact snapshot', () => {
    const { component } = setup();
    expect(component).toMatchSnapshot();
  });
});
