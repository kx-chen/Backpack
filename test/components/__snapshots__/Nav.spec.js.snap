// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Nav component should match exact snapshot 1`] = `ShallowWrapper {}`;
